src/main/java/com/minecolonies/structures/helpers/StructureProxy.java

```diff
@@ -306,8 +202,10 @@ public int getHeight()
      * Rotate the structure depending on the direction it's facing.
      *
      * @param times times to rotate.
+     * @param world the world to rotate it in.
+     * @param rotatePos the pos to rotate it around.
      */
-    public void rotate(final int times)
+    public void rotate(final int times, World world, BlockPos rotatePos)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "rotate" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "rotatePos". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)